src/LSST/M1M3/SS/Controllers/ForceController.h

```diff
@@ -195,7 +199,11 @@ class ForceController {
     void applyElevationForces();
     void zeroElevationForces();
 
+#ifdef WITH_SAL_KAFKA
+    void applyOffsetForces(std::vector<float> x, std::vector<float> y, std::vector<float> z);
```

Comment:
Please change all headers to Kafka/std::vector version. And then do the following in ../States/EngineeringState.cpp, where applyOffsetForces is called:

```
States::Type EngineeringState::applyOffsetForces(ApplyOffsetForcesCommand *command) {
    SPDLOG_INFO("{}: applyOffsetForces()", this->name);
#ifdef WITH_SAL_KAFKA
    Model::instance().getForceController()->applyOffsetForces(
            command->getData()->xForces, command->getData()->yForces, command->getData()->zForces);
#else
    Model::instance().getForceController()->applyOffsetForces(
            std::vector<float>(command->getData()->xForces, command->getData()->xForces + FA_X_COUNT ), 
            std::vector<float> (command->getData()->yForces, command->getData()->yForces + FA_Y_COUNT),
            std::vector<float> (command->getData()->zForces, command->getData()->zForces + FA_Z_COUNT);
#endif
    Model::instance().getForceController()->processAppliedForces();
    return Model::instance().getSafetyController()->checkSafety(States::NoStateTransition);
}
```

You should also pass std::vector<float> using const modifier, that might make the code a few CPU instructions shorter:

```
void applyOffsetForces(const std::vector<float> x, const std::vector<float> y, const std::vector<float> z);
```

Reply:
.. or get data pointer with data() call and you can drop vector handling:

```
States::Type EngineeringState::applyOffsetForces(ApplyOffsetForcesCommand *command) {
    SPDLOG_INFO("{}: applyOffsetForces()", this->name);
#ifdef WITH_SAL_KAFKA
    Model::instance().getForceController()->applyOffsetForces(
            command->getData()->xForces.data(), command->getData()->yForces.data(), command->getData()->zForces.data());
#else
    Model::instance().getForceController()->applyOffsetForces(
            command->getData()->xForces, command->getData()->yForces, command->getData()->zForces);
#endif
    Model::instance().getForceController()->processAppliedForces();
    return Model::instance().getSafetyController()->checkSafety(States::NoStateTransition);
}
```